tests/test_integration/test_code_generation.py

```diff
@@ -417,3 +417,32 @@ def stencil(
 
     outp.device_to_host()
     assert np.allclose(1.0, np.asarray(outp))
+
+
+# TODO(jdahm): Add additional backend support and move to test_suites
```

Comment:
If this is an easy addition I would be happier with a clean error message in case the backend does not support indirect addressing (no need in the OLD_BACKENDS anyway).

Reply:
The file name suggests that this only tests generation, but we already violate this in some places (because of the not so great `test_suits.py`). Could you also call the stencils and validate the results (one domain + `==`)?

Reply:
I cleaned up the tests so that they loop over all backends and check for an error if it's not supported. Do you think that's sufficient?